infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -865,6 +955,64 @@ private Map<UUID, StreamAddressSpace> getStreamsAddressesMap(
         return requestedAddressSpaces;
     }
 
+    /**
+     * Get a list of streams backed by which the tables have been dropped, including their CP streams
+     *
+     * @return a list of all inactive streams
+     */
+    private List<UUID> getDroppedStreams() {
+
+        if (!isRegistryTableOpened())
+        {
+            return Collections.emptyList();
+        }
+
+        CorfuRuntime corfuRuntime = getNewCorfuRuntime();
+
+        PersistentCorfuTable<CorfuStoreMetadata.TableName,
+                CorfuRecord<CorfuStoreMetadata.TableDescriptors, CorfuStoreMetadata.TableMetadata>>
+                registry = corfuRuntime.getTableRegistry().getRegistryTable();
+
+        List<UUID> realStreams = registry.entryStream()
+            .filter(entry -> entry.getValue().getMetadata().getIsDropped())
+            .map(entry -> CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(
+                entry.getKey().getNamespace(), entry.getKey().getTableName())))
+            .collect(Collectors.toList());
+        List<UUID> checkpointStreams = realStreams.stream()
+            .map(CorfuRuntime::getCheckpointStreamIdFromId)
+            .collect(Collectors.toList());
+
+        List<UUID> droppedStreams = new ArrayList<>();
+        droppedStreams.addAll(realStreams);
+        droppedStreams.addAll(checkpointStreams);
+
+        corfuRuntime.shutdown();
```

Comment:
This looks like a resource leak.

What happens if there is an exception thrown before `corfuRuntime.shutdown()` ? (Memory, Threads and file descriptors will not be freed)